hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,860 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2015 The HyperSpy developers
+#
+# This file is part of HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import division
+
+import warnings
+import numpy as np
+import math
+from contextlib import contextmanager
+
+from hyperspy.models.model1D import Model1D
+from hyperspy._signals.eds import EDSSpectrum
+from hyperspy.misc.elements import elements as elements_db
+from hyperspy.misc.eds import utils as utils_eds
+import hyperspy.components as create_component
+
+
+def _get_weight(element, line, weight_line=None):
+    if weight_line is None:
+        weight_line = elements_db[
+            element]['Atomic_properties']['Xray_lines'][line]['weight']
+    return lambda x: x * weight_line
+
+
+def _get_iweight(element, line, weight_line=None):
+    if weight_line is None:
+        weight_line = elements_db[
+            element]['Atomic_properties']['Xray_lines'][line]['weight']
+    return lambda x: x / weight_line
+
+
+def _get_sigma(E, E_ref, units_factor):
+    """
+    Gets sigma value for particular energy.
+
+    energy2sigma_factor : float
+
+    This is a factor that ???? and is set to 2.5 here with reference to
+    Goldstein ???
+
+    """
+    goldstein_factor = 2.5
+    eV2keV = 1000.
+    sigma2fwhm = 2 * math.sqrt(2 * math.log(2))
```

Comment:
Put these factors in root of the file (i.e. outside the function), at least `eV2keV` and `sigma2fwhm`, and reuse them throughout the file instead of magic numbers.